packages/engine/lib/storage/src/lib.rs

```diff
@@ -1,8 +1,8 @@
-//! Library for low-level to store and load arrow-data.
+//! Defines how data is stored in shared memory and provides an (optional) arrow-interface to it.
 //!
 //! This crates consists of two main parts:
 //!   - [`memory`] contains structures and functions to store and load data to/from shared memory.
-//!     Each [`Memory`] tracks a memory [`Segment`] consisting of four parts:
+//!     Each memory [`Segment`] consisting of four parts:
```

Comment:
I've just noticed the docs for `Memory` actually mention these four parts. Can you check that's the case? I'm not sure these docs are necessarily correct and we might need to update. I thought for Datasets (for example) it isn't an arrow batch it's just general bytes data. We _might_ still have markers but I'm not sure we have a schema... or a header

Reply:
You are correct, `DataSet` uses `Segment` and **only** `Segment` directly:
```
pub struct Dataset {
    segment: Segment,
}
```

A `Segment` is only a wrapper around `Memory` (thus it's easy to confuse those two) and a memory segment **always** keeps track for these four parts, however they are optional (simply set them to `0` length), `DataSet` sets `schema_size` and `meta_size` to `0`.

Reply:
I will make this clear in the docs as well

Reply:
I appreciate we have a todo to merge `Segment` and `Memory`, (feel free to recommit the changes you previously undid, I just wanted them in separate commits so I could review the code refactoring first), which is causing confusion. I don't quite get what you mean by this:

> a memory segment always keeps track for these four parts

Are you saying a `Segment` always keeps track of those? Or are we saying `Memory` does? Maybe we don't waste too much time discussing that specific thing if you just recommit the changes merging them which I'd prefer.

What I do want to make sure is that our docs here (probably on the `Memory` struct not necessarily the crate, I don't think we need to duplicate the info) properly outline that some of those parts are optional, as right now it seems like they're always included. (I do not count setting their length to 0 as being included). Additionally I think one of the parts is wrongly described as it's not always an Arrow RecordBatch is it?

Reply:
As `Segment` ist literally `pub struct Segment(Memory);` I would like to merge them as soon as possible (one of the very next steps). This would resolve any discussion about `Memory` and `Segment`. I don't want to do this in this PR because of #444. I would duplicate/merge the comment on `Memory` on that PR then.

Reply:
I'm fine with us doing it in a follow-up PR soon. Can we add a TODO to also re-do the docs at the same time to be clearer about how `Memory` is used for non-Arrow data.

Reply:
Added the todo to `Memory` in d6bb3be7e6199aa436b23c7f2dd242ce56cd40c7